src/main/java/com/rackspace/ceres/app/services/MetricDeletionService.java

```diff
@@ -108,91 +109,44 @@ public MetricDeletionService(ReactiveCqlTemplate cqlTemplate,
         }).then(Mono.empty());
   }
 
-  private Mono<Boolean> deleteMetric(List<Granularity> granularities, String tenant,
-      Instant timeSlot, String metricName, Flux<String> seriesSetHashes) {
-    return seriesSetHashes.flatMap(
-        seriesSetHash -> Flux.fromIterable(granularities)
-            //delete entries from downsampled tables
-            .flatMap(granularity ->
-                deleteRawOrDownsampledEntries(
-                    dataTablesStatements.downsampleDeleteWithSeriesSetHash(granularity.getWidth()),
-                    tenant, timeSlot, seriesSetHash))
-            .then(Mono.just(true))
-    )
-        .then(Mono.just(true))
-        //delete entries from raw table
-        .flatMap(item ->
-            seriesSetHashes.flatMap(seriesSetHash ->
-                deleteRawOrDownsampledEntries(dataTablesStatements.getRawDeleteWithSeriesSetHash(),
-                    tenant, timeSlot, seriesSetHash))
-                .then(Mono.just(true))
-        )
-        //delete entries from series_set_hashes table
-        .flatMap(result ->
-            seriesSetHashes.flatMap(seriesSetHash ->
-                deleteSeriesSetHashes(tenant, seriesSetHash)
-            )
-                .then(Mono.just(true)))
-        //delete entries from redis cache
-        .flatMap(result ->
-            seriesSetHashes.flatMap(seriesSetHash ->
-                removeEntryFromCache(tenant, seriesSetHash)
-            )
-                .then(Mono.just(true)))
-        //delete entries from series_sets table
-        .flatMap(result ->
-            deleteSeriesSetsByTenantIdAndMetricName(tenant, metricName)
-        );
-  }
-
   private Mono<Boolean> deleteMetricsByTenantId(List<Granularity> granularities, String tenant,
       Instant timeSlot) {
+    return deleteMetricByTenant(granularities, tenant, timeSlot)
+        .then(deleteMetadataByTenant(tenant, timeSlot));
+  }
+
+  private Mono<Boolean> deleteMetricByTenant(List<Granularity> granularities, String tenant, Instant timeSlot) {
+    return Flux.fromIterable(granularities)
+        .flatMap(granularity ->
+            deleteRawOrDownsampledEntries(dataTablesStatements
+                .downsampleDelete(granularity.getWidth()), tenant, timeSlot))
+        .then(deleteRawOrDownsampledEntries(dataTablesStatements.getRawDelete(),
+            tenant, timeSlot));
+  }
+
+
+  private Mono<Boolean> deleteMetadataByTenant(String tenant, Instant timeSlot) {
     //get series set hashes from data raw table by tenant and timeSlot
     Flux<String> seriesSetHashes = getSeriesSetHashFromRaw(tenant, timeSlot);
```

Comment:
I think we need to first get all the seriesSetHashes as this method is called after the deletion from data tables and will result empty.

Reply:
And also make sure to get series set hashes from the data table with highest granularity